src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -208,7 +213,103 @@ export const portfolioApi = createApi({
                     return isSpammyTokenText(text)
                   })
                   if (state.assets.byId[token.assetId] || isSpam) return prev
-                  prev.byId[token.assetId] = makeAsset(state.assets.byId, { ...token })
+                  let minimalAsset: MinimalAsset = token
+                  const maybePortalsAsset = maybePortalsAccounts[token.assetId]
+                  if (maybePortalsAsset) {
```

Comment:
@kaladinlight WDYT? Wondering if we should only do this for pools assets, or if the current logic of leveraging portals icons/name for all assets not present in `generatedAssetData.json` on top of unchained data makes sense? Haven't noticed any weirdnesses in my own assets list fwiw

Reply:
Idea sounds sane to me pending runtime test. Making note of the duplicate logic here. May be worth wrapping all portals things up in a little package to use throughout the repo.

Reply:
Unfortunately the package is going to be a pain (generatedAssetData.json not happy with tsc paths since it's in root vs. package path) but moved to src/utils in 7ebbe3f576

Reply:
yeah totally fine leaving it in src vs totally isolated package